.gitignore

```diff
@@ -44,3 +44,7 @@
 
 # VSCode setting
 .vscode
+
+# Docker
+/.docker
+/docker-compose.yml
```

Comment:
Die Zeile hat keine Auswirkung, da wir die Datei im Repo haben. 
gitignore wirkt sich nur auf Dateien aus, die noch nicht im Repo sind.

Reply:
Du kannst bei dir lokal mit `--assume-unchanged` arbeiten: https://dev.to/nishina555/how-to-ignore-files-already-managed-with-git-locally-19oo
Das habe ich früher hier im Repo verwendet, als wir noch nicht die `default.config.yml` hatten, sondern es nur die eine `config.yml`-Datei gab.
Wobei ich damals irgendwie immer mal Probleme mit dem Feature hatte. Und wenn man die Datei doch mal ändern will, muss man es halt erst rückgängig machen.

Reply:
Hast recht, da hab ich zu kurz gedacht.
Was haltet ihr davon, wenn wir stattdessen eine `docker-compose.yml.example` anbieten? dann hat man die probleme mit lokalen diffs gar nicht erst.

Reply:
https://docs.docker.com/compose/extends/

Dort lese ich gerade dies:

> By default, Compose reads two files, a docker-compose.yml and an optional docker-compose.override.yml file. By convention, the docker-compose.yml contains your base configuration. The override file, as its name implies, can contain configuration overrides for existing services or entirely new services.

Lässt sich damit vielleicht arbeiten? 
Wir würden dann also die `/docker-compose.override.yml` in die gitignore packen.

Reply:
Guter Hinweis, dennoch frage ich mich, ob ein Override-File für Nutzer praktikabel ist, oder ob es nicht vielleicht merklich einfacher und verständlicher wäre, das normale Compose-File zu entfernen und zu ignorieren, um anschließend ein Example-File anzubieten. Examples-Files sind ja durchaus gängig, etwa auch bei htaccess oder gitignore, und die Nutzer müssten nur noch umbenennen und könnten loslegen, ohne dass sie danach einen Diff im Repo hätten.

Override halte ich für komplizierter, richte mich aber natürlich nach euch!

Reply:
Es geht doch eigentlich nur um die Leute, die hier im Repo mitarbeiten wollen, denn die Dateien sind ja im Release sowieso nicht enthalten.
Und für die Leute, die mal schnell einen Fix machen wollen oder so, wird doch in der Regel unsere aktuelle docker-compose.yml reichen, oder? Daher finde ich es eigentlich schon praktisch, wenn die dann gar keine Datei anpassen/erstellen müssen.

Reply:
Ja, aus der Sicht ist der Weg über das zusätzliche Override-File voll in Ordnung. Ich hatte nur gedacht, man macht es den Leuten etwas einfacher, die womöglich hier mal mitarbeiten wollen.

Dann bleiben wir beim Override abschließend, was?

Reply:
Ich würde sagen ja. Aber vor allem, weil ich noch nicht verstehe, warum du den anderen Weg einfacher findest.
Im Moment haben wir in der Readme als Anleitung zum Mitarbeiten:

> You can use the provided Docker configuration to start a local development environment: docker-compose up -d.
In case you need a port other than the default port 80, start it like this: REDAXO_PORT=8080 docker-compose up -d.

Der Abschnitt wird doch komplizierter, wenn man die docker-compose.yml erst anlegen muss?
Wobei mir halt auch noch nicht klar ist, ab wann man mit der Standard-Datei nicht mehr hinkommt, und man sich dann die override-Datei anlegen werden muss.

Reply:
> Aber vor allem, weil ich noch nicht verstehe, warum du den anderen Weg einfacher findest.

Ich fand spontan den Example-Weg einfacher, weil das für mein Gefühl Best Practice ist: Wenn man den Leuten eine Vorlage an die Hand geben möchte, wie z. B. typische Rewrite Rules in der htaccess, benutzt man eine Example-Datei.

Aber du hast vollkommen recht, in diesem Fall landet es ja gar nicht mit im Release, und diese Kennzeichnung als Example-File macht man ja vorrangig aus dem Grund, damit sie an passender Stelle vorliegt, aber eben noch nicht aktiv ist. Innerhalb eines geklonten Repos aber ist es vielleicht wirklich sinnvoller, wenn solche Files direkt ausführbar vorliegen. Ich würde mich deiner Meinung inzwischen anschließen.

> Wobei mir halt auch noch nicht klar ist, ab wann man mit der Standard-Datei nicht mehr hinkommt, und man sich dann die override-Datei anlegen werden muss.

Aktuell ist das File so angelegt, dass man REDAXO mit geringem Aufwand lauffähig vor sich hat. Wenn man allerdings ernsthaft dran entwickeln möchte, möchte man vermutlich einige Dinge ergänzen, z. B. Composer vorinstallieren, E-Mailversand mit Mailhog testen, SSL-Zertifikat, Blackfire usw., etwa so wie hier: https://github.com/FriendsOfREDAXO/redaxo-mit-docker/

Mittels Override ist das alles prima möglich, ich hatte nur so ein Bauchgefühl, dass es für Nutzer unnötig kompliziert werden könnte, wenn Docker die Konfigurationen aus beiden Dateien merged. Es nimmt halt nicht nur die Override-Datei, sondern führt einen Merge durch.

Aber auch zu dem Thema bin ich inzwischen wieder anderer Meinung und stimme deinem Gedanken zu, es so umsetzen.